apps/wallet/src/ui/app/redux/slices/transactions/index.ts

```diff
@@ -104,6 +105,40 @@ export const stakeTokens = createAsyncThunk<
     }
 );
 
+type UnStakeTokenTxn = {
+    principalWithdrawAmount: string;
+    delegation: ObjectId;
+    stakedSuiId: ObjectId;
+};
+
+export const unStakeToken = createAsyncThunk<
```

Comment:
Can we move this to a `useMutation`? We shouldn't be doing this in redux anymore.

Reply:
done